examples/cookbook/architecture/command/lib/result.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Utility class to wrap result data
+///
+/// Evaluate the result using a switch statement:
+/// ```dart
+/// switch (result) {
+///   case Ok(): {
+///     print(result.value);
+///   }
+///   case Error(): {
+///     print(result.error);
+///   }
+/// }
+/// ```
+sealed class Result<T> {
+  const Result();
+
+  /// Creates an instance of Result containing a value
+  factory Result.ok(T value) => Ok(value);
```

Comment:
This can be a `const factory` to enable `const` usages of these results.

```suggestion
  const factory Result.ok(T value) = Ok;
```